flask_seasurf.py

```diff
@@ -444,29 +439,31 @@ def _before_request(self):
 
     def _after_request(self, response):
         '''
-        Checks if the `flask._app_ctx_object` object contains the CSRF token,
+        Checks if the Flask app context contains the CSRF token,
         and if the view in question has CSRF protection enabled. If both,
         goes on to check if a cookie needs to be set by verifying the cookie
         presented by the request matches the CSRF token and the user has not
         requested a token in a Jinja template.
 
-        If the token does not match or the user has requested a token,returns
+        If the token does not match or the user has requested a token, returns
         the response with a cookie containing the token. Otherwise we return
         the response unaltered. Bound to the Flask `after_request` decorator.
 
         :param response: A Flask Response object.
         '''
-        if getattr(_app_ctx_stack.top, self._csrf_name, None) is None:
+        if getattr(g, 'seasurf_csrf_token', None) is None:
             return response
 
-        _view_func = getattr(_app_ctx_stack.top, '_view_func', False)
+        _view_func = getattr(g, 'seasurf_view_func', None)
         if not (_view_func and self._should_set_cookie(_view_func)):
             return response
 
         # Don't apply set_cookie if the request included the cookie
         # and did not request a token (ie simple AJAX requests, etc)
-        csrf_cookie_matches = request.cookies.get(self._csrf_name, False) == getattr(_app_ctx_stack.top, self._csrf_name)
-        if csrf_cookie_matches and not getattr(_app_ctx_stack.top, 'csrf_token_requested', False):
+        csrf_cookie_matches = \
+            request.cookies.get(self._csrf_name, False) == g.seasurf_csrf_token
+        if csrf_cookie_matches and not getattr(g, 'seasurf_csrf_token_requested',
```

Comment:
Same here, this line can be:
`if csrf_cookie_matches and not g.seasurf_csrf_token_requested:`